homeassistant/components/hydrawise/binary_sensor.py

```diff
@@ -111,3 +132,24 @@ def available(self) -> bool:
         if self.entity_description.always_available:
             return True
         return super().available
+
+
+class HydrawiseZoneBinarySensor(HydrawiseBinarySensor):
```

Comment:
It seemed weird to have methods that are only used for binary sensors that require `self.zone` to be `not None`. The separate class made the intent clearer and resolved potential typing problems.

I don't feel terribly strongly either way, so I combined it back with the existing `HydrawiseBinarySensor` and had the service methods raise `ServiceValidationError` if `self.zone` is `None`.

Reply:
Now I noticed your comment here. So there can be both binary sensors with and w/o a zone?

Reply:
Correct. `RAIN_BINARY_SENSOR`s and `CONTROLLER_BINARY_SENSOR`s do not have zones associated with them.

Reply:
I find it strange we are adding a method to a base class that may or may not be relevant for instantiated object.  I can see why it was initially implemented in a new derived class.  That seemed like the proper modeling, rather than having a rain sensor with a method to start watering.

Reply:
In the end it doesn't really matter if they are initiated in the base class or in a subclass as it's `services.yaml` which defined which entities can be used by the service which currently is integration: hydrawise, domain: binary_sensor and device_class: running. Meaning that if any other binary sensor is providing a device class of running the user can still pick it and it will not work as self.zone doesn't exist.
Is that the case then we need to handle it, if not then we're all good (looks like we don't need to handle it).

So I think either you can model it like you did previously with a subclass or you can replace the if-statement with an assert instead. Maybe a subclass is clearer and also can have a good comment about it's purpose. 👍

Reply:
Okay, added back the separate class with an updated class docstring.